Modules/Microsoft365DSC/DSCResources/MSFT_TeamsShiftsPolicy/MSFT_TeamsShiftsPolicy.psm1

```diff
@@ -0,0 +1,476 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.Int64]
+        $AccessGracePeriodMinutes,
+
+        [Parameter()]
+        [System.String]
+        $AccessType,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableScheduleOwnerPermissions,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableShiftPresence,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeFrequency,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeMessageCustom,
+
+        [Parameter()]
+        [System.String]
+        $ShiftNoticeMessageType,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    try
+    {
+        New-M365DSCConnection -Workload 'MicrosoftTeams' `
+            -InboundParameters $PSBoundParameters | Out-Null
+    }
+    catch
+    {
+        Write-Verbose -Message $_
```

Comment:
Why do you only write a verbose here? If the catch is triggered, doesn't that mean creating the connection failed? And in that case, shouldn't the rest of the code be aborted?

Reply:
To be aligned with the other resources, we will remove the try/catch. The error will be trapped inside the New-M365DSCCOnnection function instead